src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/compute/v2/AwsV2ComputePlugin.java

```diff
@@ -1,47 +1,459 @@
+		AwsHardwareRequirements flavor = null;
+
+		Map<String, Integer> imagesMap = getImagesMap(cloudUser);
+		for (Entry<String, Integer> images : imagesMap.entrySet()) {
+			for (String line : lines) {
+				if (!line.startsWith(COMMENTED_LINE_PREFIX)) {
+					requirements = line.split(CSV_COLUMN_SEPARATOR);
+					flavor = mountHardwareRequirements(images, requirements);
+					this.flavors.add(flavor);
+				}
+			}
+		}
+	}
+
+	protected AwsHardwareRequirements mountHardwareRequirements(Entry<String, Integer> image, String[] requirements) {
+		String name = requirements[INSTANCE_TYPE_COLUMN];
+		String flavorId = getRandomUUID();
+		int cpu = Integer.parseInt(requirements[VCPU_COLUMN]);
+		Double memory = Double.parseDouble(requirements[MEMORY_COLUMN]) * ONE_GIGABYTE;
+		int disk = image.getValue();
+		String imageId = image.getKey();
+		Map<String, String> requirementsMap = loadRequirementsMap(requirements);
+		return new AwsHardwareRequirements(name, flavorId, cpu, memory.intValue(), disk, imageId, requirementsMap);
+	}
+
+	protected Map<String, String> loadRequirementsMap(String[] requirements) {
+		Map<String, String> requirementsMap = new HashMap<String, String>();
+		String processor = requirements[PROCESSOR_COLUMN];
+		String performance = requirements[NETWORK_PERFORMANCE_COLUMN];
+		String storage = requirements[STORAGE_COLUMN];
+		String bandwidth = requirements[DEDICATED_EBS_BANDWIDTH_COLUMN];
+		requirementsMap.put(PROCESSOR_REQUIREMENT, processor);
+		requirementsMap.put(PERFORMANCE_REQUIREMENT, performance);
+		requirementsMap.put(STORAGE_REQUIREMENT, storage);
+		requirementsMap.put(BANDWIDTH_REQUIREMENT, bandwidth);
+		return requirementsMap;
+	}
+
+	protected Map<String, Integer> getImagesMap(AwsV2User cloudUser)
+			throws InvalidParameterException, UnexpectedException {
+
+		Map<String, Integer> imageMap = new HashMap<String, Integer>();
+		DescribeImagesRequest request = DescribeImagesRequest.builder()
+				.owners(cloudUser.getId())
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		DescribeImagesResponse response = client.describeImages(request);
+
+		List<Image> images = response.images();
+		for (Image image : images) {
+			int size = getImageSize(image);
+			imageMap.put(image.imageId(), size);
+		}
+		return imageMap;
+	}
+
+	protected List<String> loadLinesFromFlavorFile() throws ConfigurationErrorException {
+		Path path = Paths.get(this.flavorsFilePath);
+		try {
+			return Files.readAllLines(path);
+		} catch (IOException e) {
+			LOGGER.error(String.format(Messages.Error.ERROR_MESSAGE, e), e);
+			throw new ConfigurationErrorException();
+		}
+	}
+
+	protected int getImageSize(Image image) {
+		int size = 0;
+		for (BlockDeviceMapping device : image.blockDeviceMappings()) {
+			size = device.ebs().volumeSize();
+		}
+		return size;
+	}
+	
+	// This method is used to aid in the tests
+	protected void setFlavorsFilePath(String flavorsFilePath) {
```

Comment:
why don't u do that in the constructor?

Reply:
I do but I tried to use this method to perform a test.
I changed the method to get and mock the result in the test.